src/api/search_transactions.rs

```diff
@@ -2,75 +2,83 @@ use coinbase_mesh::models::{
   AccountIdentifier, BlockIdentifier, BlockTransaction, SearchTransactionsRequest, SearchTransactionsResponse,
   Transaction, TransactionIdentifier,
 };
+use convert_case::{Case, Casing};
 use serde_json::json;
 use sqlx::FromRow;
 
 use crate::{
-  operation, util::DEFAULT_TOKEN_ID, ChainStatus, MinaMesh, MinaMeshError, OperationType, TransactionStatus,
-  UserCommandType,
+  operation, util::DEFAULT_TOKEN_ID, ChainStatus, InternalCommandType, MinaMesh, MinaMeshError, OperationType,
+  TransactionStatus, UserCommandType,
 };
 
 impl MinaMesh {
   pub async fn search_transactions(
     &self,
     req: SearchTransactionsRequest,
   ) -> Result<SearchTransactionsResponse, MinaMeshError> {
-    let user_commands = self.fetch_user_commands(&req).await?;
-    let user_commands_len = user_commands.len();
-    let next_offset = req.offset.unwrap_or(0) + user_commands_len as i64;
+    let original_offset = req.offset.unwrap_or(0);
+    let mut offset = original_offset;
+    let mut limit = req.limit.unwrap_or(100);
+    let mut transactions = Vec::new() as Vec<BlockTransaction>;
```

Comment:
The `as Vec<BlockTransaction>` shouldn't be necessary / should be inferred. If it is not for some reason, then perhaps `Vec::new::<BlockTransaction>()` instead.